drivers/syslog/syslog_flush.c

```diff
@@ -64,21 +64,27 @@
 
 int syslog_flush(void)
 {
-  DEBUGASSERT(g_syslog_channel != NULL);
+  int i;
+
+  for (i = 0; i < CONFIG_SYSLOG_MAX_CHANNELS; i++)
+    {
+      if (g_syslog_channel[i] == NULL)
+        break;
 
 #ifdef CONFIG_SYSLOG_INTBUFFER
-  /* Flush any characters that may have been added to the interrupt
-   * buffer.
-   */
+      /* Flush any characters that may have been added to the interrupt
+       * buffer.
+       */
 
-  syslog_flush_intbuffer(g_syslog_channel, true);
+      syslog_flush_intbuffer(g_syslog_channel[i], true);
```

Comment:
We just have one interrupt buffer, so the loop need move into syslog_flush_intbuffer otherwise, all channel except the first one lose the log happen in the interrupt handler.

Reply:
Correct, I didn't notice this. Fixed.